### PR TITLE
new extension: attributes ... allows attribute-only swap 

### PR DESCRIPTION
**Pull request by stukennedy**
- hx-swap and hx-swap-oob to have an "attributes" value
- swap the attributes of the target element and not its children
- ensure hyperscript is processed after swap

## Description
There are many occasions in a deeply-nested DOM structure where attribute-only changes are desirable.
In the spirit of HATEOAS it is desirable that the back-end drives the state and behaviour changes in the front-end.
Even simple class changes or aria attribute changes are better delivered from the back-end than programmed in the client. 
But the power of this approach comes in the ability to be able to make targeted changes to HTMX attributes, allowing for branching behaviours without needing to return entire components, which can become very messy, especially if targeting multiple OOB components that have children.

Example of changing form behaviour:
```html
<form hx-post="/original-action">
    <div>Bunch of stuff</div>
    <input name="foo" type="text" />
    <button type="submit">Submit</button>
    <a hx-post="/change-form" hx-swap="attributes" hx-target="form">Change form action</a>
</form>
```
server responds with
```html
<form  hx-post="/new-action"></form>
```

(This extension will not override the `id` or the `hx-ext` attribute of the target.)

Corresponding issue:

## Testing
test file added to test new behaviour and test existing behaviour is unaltered by the extension

## Checklist

* [x] I have read the contribution guidelines
* [x] I have targeted this PR against the correct branch (`master` for website changes, `dev` for
  source changes)
* [x] This is either a bugfix, a documentation update, or a new feature that has been explicitly
  approved via an issue
* [x] I ran the test suite locally (`npm run test`) and verified that it succeeded

### CODE REVIEW

**Comment by Telroshan** [comment]
Hey, I would suggest publishing your own extension repo for now and list it in this repo's community extensions.
Adding an extension to the repo itself makes it kind of an "official" extension that we'd have to provide long-term support for, I hope you'll understand we can't do that with all extensions!
All extensions in this repo were already there in 1.0, and with htmx 2 we wanted to take a different approach, to let community easily register their extensions in our extensions list while we let them handle their support.

We can always discuss later if an extension should or should not make it into the core, but I would suggest making it a community extension first.

I know this can feel frustrating, apologies for that! We'll have to define contributing guidelines at some point for this repo, we're still playing it by ear for now

_Note: yes, this is a shameless copy-paste from my comment in #44, but it applies just as well!_

File: src/attributes/attributes.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+					}
+				} else {
+					this.swapAttributes(target, source);
+				}
+				return true;
+			}
+			return false;
+		},
+
+		swapAttributes: function (target, source) {
```
Comment (MichaelWest22) [inline src/attributes/attributes.js:22]: Hi stuckennedy I think this function should not be returned as a base element of the extension definition as this is designed really only for the 6 or so overridable functions and while it will probably not break anything it can add confusion.  This function could be defined with inline scope just like the forEach function inside it does or moved out of the scope of the extension definition to the top maybe.

File: src/attributes/attributes.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+(function () {
+	htmx.defineExtension('attributes', {
+		handleSwap: function (swapStyle, target, fragment) {
+			var source = fragment.firstElementChild || fragment;
```
Comment (MichaelWest22) [inline src/attributes/attributes.js:4]: I was trying to work out what this firstElementChild alternative source option here was trying to do and why you are checking for hx-swap-oob and doing a actualTarget lookup.  I don't think i needed this when I was testing oob attribute swaps with my earlier test version i did as the handleSwap is called once for main swaps and then again for each oob swap and so it shouldn't really need extra handling logic.  But maybe you found an edge case you needed to handle
